src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supply = calculateSupply(fromFileName);
+        int buy = calculateBuy(fromFileName);
+        int result = supply - buy;
+        writeToFile(toFileName, supply, buy, result);
+        printToConsole(supply, buy, result);
+    }
+
+    private int calculateSupply(String fileName) {
+        return calculate(fileName, SUPPLY);
+    }
+
+    private int calculateBuy(String fileName) {
+        return calculate(fileName, BUY);
+    }
+
+    private int calculate(String fileName, String operation) {
+        int total = 0;
```

Comment:
It's inefficient to read the same file, and do calculations twice when you can do it in one go, please refactor. Your code should demonstrate a clear separation of concerns - separate method for reading, preparing data, and writing prepared data to the file (in one go, don't construct report string in writer method)